src/analysis/processing/qgsalgorithmrandompointsonlines.cpp

```diff
@@ -0,0 +1,264 @@
+/***************************************************************************
+                         qgsalgorithmrandompointsonlines.cpp
+                         ---------------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Håvard Tveite
+    email                : havard dot tveite at nmbu dot no
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsalgorithmrandompointsonlines.h"
+#include "random"
+
+///@cond PRIVATE
+
+QString QgsRandomPointsOnLinesAlgorithm::name() const
+{
+  return QStringLiteral( "randompointsonlines" );
+}
+
+QString QgsRandomPointsOnLinesAlgorithm::displayName() const
+{
+  return QObject::tr( "Random points on lines" );
+}
+
+QStringList QgsRandomPointsOnLinesAlgorithm::tags() const
+{
+  return QObject::tr( "random,points,lines,seed,attributes,create" ).split( ',' );
+}
+
+QString QgsRandomPointsOnLinesAlgorithm::group() const
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsRandomPointsOnLinesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsRandomPointsOnLinesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input line layer" ), QList< int >() << QgsProcessing::TypeVectorLine ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "POINTS_NUMBER" ), QObject::tr( "Number of points for each feature" ), QgsProcessingParameterNumber::Integer, 1, false, 1 ) );
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "MIN_DISTANCE" ), QObject::tr( "Minimum distance between points" ), 0, QStringLiteral( "TARGET_CRS" ), true, 0 ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "TARGET_CRS" ), QObject::tr( "Target CRS" ), QStringLiteral( "ProjectCrs" ), false ) );
+
+  //addParameter( new QgsProcessingParameterNumber( QStringLiteral( "MAX_TRIES_PER_POINT" ), QObject::tr( "Maximum number of search attempts (relevant when Minimum distance between points > 0)" ), QgsProcessingParameterNumber::Integer, 10, false, 1 ) );
```

Comment:
Ok, so native Processing algorithms are entirely designed around the principal of being as simple and modular as possible (see also note above about the crs param). We try to avoid the massive overload of options in these native algorithms which you see in the 3rd party ones (like grass and saga). The advanced group is one way to assist with this -- anything which falls into the "most users would not need to change this setting" OR "this setting could be confusing for users" should be placed in the advanced group. 

* It's not 100% consistent, so if you do identify parameters you think are good candidates to move over, please open PRs or file issues!